src/types.db

```diff
@@ -231,6 +231,12 @@ serial_octets           rx:DERIVE:0:U, tx:DERIVE:0:U
 signal_noise            value:GAUGE:U:0
 signal_power            value:GAUGE:U:0
 signal_quality          value:GAUGE:0:U
+slurm_job_state         value:GAUGE:0:U
+slurm_stats_load        server_thread_count:GAUGE:0:U, agent_thread_count:GAUGE:0:U, agent_queue_size:GAUGE:0:U, dbd_agent_queue_size:GAUGE:0:U
+slurm_node_state        value:GAUGE:0:U
+slurm_stats_backfill    backfilled_jobs:DERIVE:0:U, backfilled_pack_jobs:DERIVE:0:U, cycle_counter:COUNTER:0:U, cycle_last:GAUGE:0:U, cycle_max:COUNTER:0:U, cycle_sum:COUNTER:0:U, cycle_last_depth:GAUGE:0:U, cycle_last_depth_try:GAUGE:0:U, depth_sum:COUNTER:0:U, depth_try_sum:COUNTER:0:U, queue_len:GAUGE:0:U, queue_len_sum:COUNTER:0:U
```

Comment:
This also mixes all-into-one, but in different form.
The goal was not in replacing `type_instance` into `value` index.

Do you really found all these metrics useful?
I found no documentation about these values, so there is no reasons to report all them.

From my POV we do engineering, so we should argue all things.
If there is no such ready documentation, please fill a table with all values you have from `slurm_get_statistics()` call,
add explanation about each that value and that will argue filling subsequent columns  by `type` and `type_instance` values.

That is the thing should be done before writing the code.

Reply:
Unfortunately the slurm documentation of the api is very lacking, but the best you can find is the [sdiag](https://github.com/SchedMD/slurm/blob/master/doc/man/man1/sdiag.1) manpage and [sdiag.c](https://github.com/SchedMD/slurm/blob/master/src/sdiag/sdiag.c) source.

I will add a reference to these in the collectd manpage and in the slurm.c module as a comment, as this would make the meaning of each metric easier to look up.

All these metrics are useful and needed, this is why the official slurm diagnosis tool reports all of those that I have included. (In my particular case some of these metrics may be constant and do not vary much or at all, but this is not a reason for excluding them, as they may be more useful for other slurm clusters.)

In theory metrics that end in _sum or _max could be derived from _last* kind of metrics, but in practice I found that these usually diverge, so keeping them seems sensible. Also, it may make it easier to have a global mean just like sdiag does by computing sum / count.

I will propose a new type listing and we can discuss. Would it be ok to re-use other types, e.g. `threads` for the `server_thread_count`, `queue_length` for the `agent_queue_size`, etc? Or is it preferred to duplicate the metric type with something like slurmctld_thread_count, slurm_agent_queue_size, etc?

Reply:
>Would it be ok to re-use other types, e.g. threads for the server_thread_count, queue_length for the agent_queue_size, etc? 

That would be ok.

Reply:
>In theory metrics that end in _sum or _max could be derived from _last* kind of metrics, but in practice I found that these usually diverge, so keeping them seems sensible.

I think `max` has no sence, as it will never decrease, until restart.
So, it is better to look for specified interval of `_last`, if that is needed.

Also, I think possibly `_sum` should go to some DERIVE type, but this not a rule, but an opinion.

Reply:
Thank you a lot for your fast and helpful responses.

> Also, I think possibly `_sum` should go to some DERIVE type, but this not a rule, but an opinion.

Yes, I think this makes sense. Possibly COUNTER, although it may get reset (a condition different from overflow).

I have been experimenting with DERIVE for slurm_stats_jobs and I notice something unexpected: While gauge values will display the following values for one of the metrics:
```
1559136017.745,41.000000
1559136077.575,42.000000
1559136137.578,42.000000
1559136197.575,43.000000
```
changing the same metric to DERIVE or COUNTER will provide the following values:
```
1559136416.574,0.000000
1559136476.575,2345593565836.732910
```
Is this expected? (This is output from the CSV plugin, and I see the same values on my grafana dashboards) It seems like an increment of 1 for this metric is causing the derivative to jump from 0 to a very high value, while I was expecting a value of 1. I may be misunderstanding something here, but wanted to clarify before switching everything to DERIVE types.

Reply:
> It seems like an increment of 1 for this metric is causing the derivative to jump from 0 to a very high value, while I was expecting a value of 1. 

If I understand correctly, when you possibly has to change GAUGE to DERIVE you had to restart all services, as there are internal states.

Reply:
>but wanted to clarify before switching everything to DERIVE types.

If value monotonously growing, that is candidate to DERIVE.